api/app/reviews/tests.py

```diff
@@ -1518,6 +1518,50 @@ def test_get_reference_not_yet_submitted(self):
         self.assertEqual(len(data), 6)
         self.assertEqual(len(data['references']), 0)
 
+    def test_get_reference_submitted_later(self):

```

Comment:
@greg-newman 
This is the test :)
This [link](https://docs.sqlalchemy.org/en/13/errors.html#error-bhk3) made a lot of sense, so I'm not sure now if my test is dodgy:) 
This is the error:

`======================================================================
ERROR: In this test, a reference has been requested, but the reference is only submitted after the second check
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/code/api/app/reviews/tests.py", line 1540, in test_get_reference_submitted_later
    'token': reference_req.token,
  File "/env/local/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py", line 287, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "/env/local/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py", line 718, in get
    value = state._load_expired(state, passive)
  File "/env/local/lib/python2.7/site-packages/sqlalchemy/orm/state.py", line 652, in _load_expired
    self.manager.deferred_scalar_loader(self, toload)
  File "/env/local/lib/python2.7/site-packages/sqlalchemy/orm/loading.py", line 946, in load_scalar_attributes
    "attribute refresh operation cannot proceed" % (state_str(state))
DetachedInstanceError: Instance <ReferenceRequest at 0x7f0df593b990> is not bound to a Session; attribute refresh operation cannot proceed (Background on this error at: http://sqlalche.me/e/13/bhk3)
`

Reply:
Hey! Sorry for the drive-by comment, is this error coming from line 1540? We see this error very often in our tests, it's something wrong with our setup and/or a bug in the version of SQLAlchemy we're on. Basically, if you try to access a property of a database-connected object *after* making a call to the API, you get this error. So after you've done a self.app.get (or post or put), you can't access properties on database objects anymore. The work-around is to access the properties before doing any API calls, so if you move lines 1539 to 1542 before line 1530 it should work.

Reply:
Would be so amazing If you can figure out how to fix it :D

Reply:
@avishkar58 Thank you! :D 
I definitely remember this issue coming up when we were working together, I just wanted to be sure before falling on that!

Reply:
I wonder if we could globally disable lazy loading for tests?

Reply:
Thanks @avishkar58!